hw/ip/pinmux/doc/_index.md

```diff
@@ -6,122 +6,383 @@ title: "Pinmux Technical Specification"
 - Programmable mapping from top-level inputs to peripheral inputs
 
-## Description
+- Programmable control of chip pad attributes like output drive-strength, pull-up, pull-down and virtual open-drain
+
+- Programmable pattern detectors to detect wakeup conditions during sleep mode
+
+- Programmable sleep mode behavior
+
+- Support for life-cycle-based JTAG (TAP) isolation and muxing
 
-The pinmux peripheral is a programmable module designed to wire arbitrary peripheral inputs and outputs to arbitrary multiplexable chip bidirectional pins.
+# Theory of Operations
+
+## Block Diagram and Overview
+
+The `pinmux` peripheral is a programmable module designed to wire arbitrary peripheral inputs and outputs to arbitrary multiplexable chip bidirectional pins.
 It gives much flexibility at the top level of the device, allowing most data pins to be flexibly wired and controlled by many peripherals.
-It is assumed that all available pins that the pinmux connects to are bidirectional, controlled by logic within this module.
-This document does not define how these are connected to pads at the toplevel, since that is governed by the `padctrl` IP (**TODO**).
-However, some discussion is shared in a later section.
+Even though the `pinmux` is referred to as one IP, it is logically split into two modules that are instantiated on the top-level and the chip-level, respectively, as can be seen in the block diagram below.
+The top-level module `pinmux` contains the CSRs accessible via the TL-UL interface, the main muxing matrix, retention registers, a set of programmable wakeup detectors, and the HW strap sampling and TAP / JTAG muxing logic.
+The chip-level module `padring` instantiates the bidirectional pads and connects the physical pad attributes.
+
+![Pinmux Block Diagram](pinmux_overview_block_diagram.svg)
 
-The number of available peripheral IOs and muxed IOs is configurable, in other words modifiable at design time.
-This configurability is implemented by representing inputs / outputs as packed arrays, in combination with the SystemVerilog parameters `NPeriphIn`, `NPeriphOut` and `NMioPads`.
+### MIO and DIO Signal Categories
+
+The `pinmux` supports two different IO signal categories:
+Muxed IO (MIO) signals that are routed through the `pinmux` matrix, and dedicated IO (DIO) signals that bypass the `pinmux` matrix.
+This allows to accommodate IO signals that are timing critical or that must have a fixed IO mapping for another reason.
+Note that although DIO signals are not routed through the `pinmux` matrix, they are still connected to the retention logic and the wakeup detectors (see next section below).
+
+The number of available peripheral IOs and pads and their assignment to the MIO / DIO categories is done at design time as part of the top-level configuration.
+This configurability is achieved by representing inputs / outputs as packed arrays, in combination with the SystemVerilog parameters `NPeriphIn`, `NPeriphOut` and `NMioPads` and `NDioPads`.
```

Comment:
parameters `NPeriphIn`, `NPeriphOut` and `NMioPads` and `NDioPads` --> parameters `NPeriphIn`, `NPeriphOut`, `NMioPads` and `NDioPads`

Reply:
fixed